File: src/main/scala/ru/albemuth/analysis/population/v3/Man.scala
package ru.albemuth.analysis.population.v3

import ru.albemuth.analysis.population.v3.Human.Sex

/**
 * @author vovan
 * Класс, моделирующий мужчины
 */
class Man(aCranialDiameter: Double, anAge: Int, anAgeLimit: Int) extends Human(aCranialDiameter, anAge, anAgeLimit) {

    val sex = Sex.MALE

    def isFatherhoodAccepted: Boolean = {
        //отцовство возможно для мужчин от 17 до 33-х лет
        17 <= age && age <= 33
    }

}

File: src/main/java/ru/albemuth/analysis/population/v1/Man.java
package ru.albemuth.analysis.population.v1;

/**
 * @author vovan
 * Класс, моделирующий мужчину
 */
public class Man extends Human {

    public Man(int ageLimit, double firstLaborDeathP) {
        super(ageLimit, firstLaborDeathP);
    }

    @Override
    public Sex getSex() {
        return Sex.MALE;
    }

    public boolean isFatherhoodAccepted() {
        return 17 <= age && age <= 33;//отцовство возможно для мужчин от 17 до 33-х лет
    }

}
